app/src/main/java/com/example/plantsapp/data/firebase/repository/FirebaseUserRepository.kt

```diff
@@ -0,0 +1,21 @@
+package com.example.plantsapp.data.firebase.repository
+
+import com.example.plantsapp.domain.model.User
+import com.example.plantsapp.domain.repository.UserRepository
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class FirebaseUserRepository @Inject constructor() : UserRepository {
+
+    override var user: User? = null
```

Comment:
If you are not using public val user that is nullable - maybe just remove it from interface